src/orion/core/utils/__init__.py

```diff
@@ -180,3 +182,25 @@ def __call__(cls, of_type, *args, **kwargs):
         error += "\nCurrently, there is an implementation for types:\n"
         error += str(sorted(cls.types.keys()))
         raise NotImplementedError(error)
+
+
+# pylint: disable = unused-argument
+def _handler(signum, frame):
+    log.error("Oríon has been interrupted.")
+    raise KeyboardInterrupt
+
+
+@contextmanager
+def sigterm_as_interrupt():
+    """Intercept ``SIGTERM`` signals and raise ``KeyboardInterrupt`` instead"""
+    ## Signal only works inside the main process
+    previous = None
+    try:
+        previous = signal.signal(signal.SIGTERM, _handler)
+    except ValueError:
```

Comment:
Do we want to silence the error if `signal.signal` is called in a child process? It seems to me we want to know because we should not rely on sigterm_as_interrupt in this situation.

Reply:
Done.